core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -278,6 +281,62 @@ static ByteStreamMessage of(Path path, @Nullable ExecutorService executor, ByteB
         return builder.alloc(alloc).bufferSize(bufferSize).build();
     }
 
+    /**
+     * Creates a new {@link ByteStreamMessage} that publishes {@link HttpData}s from the specified
+     * {@linkplain Consumer outputStreamConsumer}.
+     *
+     * <p>For example:<pre>{@code
+     * ByteStreamMessage byteStreamMessage = StreamMessage.fromOutputStream(os -> {
+     *     try (os) {
+     *         for (int i = 0; i < 5; i++) {
+     *             os.write(i);
+     *         }
+     *     } catch (IOException e) {
+     *         throw new RuntimeException(e);
+     *     }
+     * });
+     * byte[] result = byteStreamMessage.collectBytes().join();
+     *
+     * assert Arrays.equals(result, new byte[] { 0, 1, 2, 3, 4 });
+     * }</pre>
+     */
```

Comment:
How about not using the try-with-resources statement in the example and add description?
```suggestion
     * ByteStreamMessage byteStreamMessage = StreamMessage.fromOutputStream(os -> {
     *     try {
     *         for (int i = 0; i < 5; i++) {
     *             os.write(i);
     *         }
     *         os.close();
     *     } catch (IOException e) {
     *         throw new RuntimeException(e);
     *     }
     * });
     * byte[] result = byteStreamMessage.collectBytes().join();
     *
     * assert Arrays.equals(result, new byte[] { 0, 1, 2, 3, 4 });
     * }</pre>
     * 
     * <p>Please note that the try-with-resources statement is not used to call {@code os.close()}
     * automatically. It's because the {@link OutputStream} is closed by the {@link StreamMessage} and
     * the exception is propagated to the {@link Subscriber} automatically.
     */
```

Reply:
nit: Perhaps, `UncheckedIOException` would be better to convert an IOException to a runtime exception.

Reply:
✅ updated, thanks for the nice suggestion!